docs/source/api/rasterio-udfs.rst

```diff
@@ -234,40 +234,118 @@ Firstly we will create a spark DataFrame from a directory of raster files.
+        extensions_map = rasterio.drivers.raster_driver_extensions()
+        driver_map = {v: k for k, v in extensions_map.items()}
+        extension = driver_map[driver] #e.g. GTiff
+        file_name = f"{file_id}.{extension}"
+
+        # - [3] write local raster
+        # - this is showing a single band [1]
+        #   being written
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = f"{tmp_dir}/{file_name}"
+            # here you can update profile using .update method
+            # example https://rasterio.readthedocs.io/en/latest/topics/writing.html
+            with rasterio.open(
+                tmp_path,
+                "w",
+                **profile
+            ) as dst:
+                dst.write(raster,1) # <- adjust as needed
+
+            # - [4] copy to fuse path
+            Path(fuse_dir).mkdir(parents=True, exist_ok=True)
+            fuse_path = f"{fuse_dir}/{file_name}"
+            if not os.path.exists(fuse_path):
+                shutil.copyfile(tmp_path, fuse_path)
+        return fuse_path
+
+Finally we will apply the function to the DataFrame.
+
+.. code-block:: python
+
+    df.select(
+      write_raster(
+        "tile.raster",
+        lit("GTiff").alias("driver"),
+        "uuid",
+        lit("dbfs:/path/to/output/dir").alias("fuse_dir")
+      )
+    ).display()
+    +-------------------------------------+
+    | write_raster(raster, driver, uuid, fuse_dir)|
+    +-------------------------------------+
+    | dbfs:/path/to/output/dir/1234.tif   |
+    | dbfs:/path/to/output/dir/4545.tif   |
+    | dbfs:/path/to/output/dir/3215.tif   |
+    | ...                                 |
+    +-------------------------------------+
+
+Sometimes you don't need to be quite as fancy. Consider when you simply want to specify to write out raster contents,
+in this case as a TIF (or you could specify the extension or include it in the file_id).
+
+.. code-block:: python
 
     @udf("string")
-    def write_raster(raster, file_id, parent_dir):
-      with MemoryFile(BytesIO(raster)) as memfile:
-        with memfile.open() as dataset:
-          Path(outputpath).mkdir(parents=True, exist_ok=True)
-          extensions_map = rasterio.drivers.raster_driver_extensions()
-          driver_map = {v: k for k, v in extensions_map.items()}
-          extension = driver_map[dataset.driver]
-          path = f"{parent_dir}/{file_id}.{extension}"
-          # If you want to write the raster to a different format
-          # you can update the profile here. Note that the extension
-          # should match the driver format
-          with rasterio.open(path, "w", **dataset.profile) as dst:
-            dst.write(dataset.read())
-          return path
+    def write_tif(raster, file_id, fuse_dir):
+        from pathlib import Path
+        import os
+        import shutil
+        import tempfile
+
+        Path(fuse_dir).mkdir(parents=True, exist_ok=True)
+        file_name = f"{file_id}.tif"
```

Comment:
@mjohns-databricks we cannot decide this is a tif.
This should have the extensions bit as the previous method as well.

Reply:
I change it to more generic `write_content` which makes it more clear on the content (no .tif involved also).